### PR TITLE
Adding FCS online gain used on DEP in offline DB

### PR DESCRIPTION
**Pull request by akioogawa**
Adding FCS online gain used on DEP in offline DB which will be used for offline trigger emulation (StFcsTriggerSimMaker). Offline DB already has tables pushed. Here is code needed to reading of those tables, and using them. Assorted macros to push those into DB and input files are also in here for future reference/updates. Those gainOnline tables will be only useful for trigger emulation in offline, and other analysis only uses gain and gaincorr for offline energy gain calibrations.

### CODE REVIEW

File: StRoot/StFcsDbMaker/StFcsDb.cxx [hunk 1]
```diff
@@ -354,8 +373,31 @@ int StFcsDb::InitRun(int runNumber) {
...
+	    getDepfromId(det,id,ehp,ns,crt,slt,dep,ch);
+	    mGainOnline[ehp][ns][dep][ch]=mFcsPresThreshold.threshold[ip];
+	    ip++;
+        }
+    }    
+
     // Get beamline 
-    TDataSet* dbDataSet = StMaker::GetChain()->GetDataBase("Calibrations/rhic/vertexSeed");
+    //TDataSet* dbDataSet = StMaker::GetChain()->GetDataBase("Calibrations/rhic/vertexSeed");
+    TDataSet* dbDataSet = 0;
```
Comment (mrosales2016) [inline StRoot/StFcsDbMaker/StFcsDb.cxx:400]: Hi Akio, could you please explain the redefinition of the dbDataSet variable here. With this definition the block of code below would not execute. I also don't see this in the code that I copied over from your local directory so I was curious about this change.

**Comment by akioogawa** [comment]
Please someone review this?

**Review by veprbl — APPROVED** [review]
Manny says this is good. The question about vertex db unset was marked as resolved by @akioogawa without a response. I assume that is okay, and the change acceptable.

**Review by mrosales2016 — APPROVED** [review]

File: StRoot/StFcsDbMaker/StFcsDb.cxx (hunk 1 above)
Comment (akioogawa) [reply StRoot/StFcsDbMaker/StFcsDb.cxx:400]: This is not currenty used but here as placeholder for when significant crossing angle exist (like pAu, planned but not happening in run24). This cuased some crashes and this is walk around. Yes this looks bad but we need to resolve this when we needed.

**Comment by akioogawa** [comment]
Sorry somehow my comment was not recorded and resolved without comments. I now added back my comment. Need one more reviwer please.

File: StRoot/StFcsDbMaker/StFcsDb.cxx (hunk 1 above)
Comment (mrosales2016) [reply StRoot/StFcsDbMaker/StFcsDb.cxx:400]: Great that addresses my question thank you for the update.

**Comment by veprbl** [comment]
@plexoos , please

**Review by zlchang — APPROVED** [review]

**Pull request merged**